### PR TITLE
Finish Graph bert modeling

### PR DESCRIPTION
**Pull request by L1aoXingyu**
完成基于 Lazy Bert 的 Graph Bert 模型搭建，完成训练脚本和推理脚本

## 训练 loss 对齐
对齐 1000 iters loss，数据集使用 https://github.com/Oneflow-Inc/OneFlow-Benchmark/tree/master/LanguageModeling/BERT 中的[样本数据集](https://oneflow-public.oss-cn-beijing.aliyuncs.com/datasets/wiki_ofrecord_seq_len_128_example.tgz)，关掉 dataloader random shuffle，graph 模型载入 lazy 模型的 weight 作为初始化

实验配置
batch size: 32
iter_num: 1000
lr: 1e-3
scheduler: None
optimizer：Adam
前 1000 iters 和 lazy 的 loss 曲线如下
lazy_graph_loss

## inference 对齐
载入 lazy 模型进行一次前向推理，graph 和 lazy 获得相同的 outputs 和 loss，相关验证代码 https://github.com/Oneflow-Inc/models/blob/8beb2e73395db5943e627421c3d984c83e4e11ef/bert-oneflow/compare_lazy_outputs.py#L142-L159

### CODE REVIEW

File: bert-oneflow/modeling.py [hunk 1]
```diff
@@ -0,0 +1,442 @@
...
+        hidden_size,
+        hidden_dropout_prob,
+        seq_length,
+    ):
+        super().__init__()
+        self.word_embeddings = nn.Embedding(vocab_size, hidden_size)
+        self.position_embeddings = nn.Embedding(max_position_embeddings, hidden_size)
+        self.token_type_embeddings = nn.Embedding(type_vocab_size, hidden_size)
+
+        self.LayerNorm = nn.LayerNorm(hidden_size)
```
Comment (MARD1NO) [inline bert-oneflow/modeling.py:24]: self.LayerNorm -> self.layernorm好点

**Comment by L1aoXingyu** [comment]
load lazy 参数以及和 lazy 对齐的代码已经被清理掉了，在这个 pr 中做了备份 https://github.com/Oneflow-Inc/models/pull/117

File: bert-oneflow/modeling.py [hunk 2]
```diff
@@ -0,0 +1,442 @@
...
+            context_layer, [-1, self.seq_len, self.all_head_size]
+        )
+        return context_layer
+
+
+class BertSelfOutput(nn.Module):
+    def __init__(self, hidden_size, hidden_dropout_prob):
+        super().__init__()
+        self.dense = nn.Linear(hidden_size, hidden_size)
+        self.LayerNorm = nn.LayerNorm(hidden_size)
```
Comment (MARD1NO) [inline bert-oneflow/modeling.py:105]: 同上

File: bert-oneflow/modeling.py [hunk 3]
```diff
@@ -0,0 +1,442 @@
...
+    def __init__(
+        self,
+        num_attention_heads,
+        hidden_size,
+        seq_len,
+        hidden_dropout_prob=0.1,
+        attention_probs_dropout_prob=0.1,
+    ):
+        super().__init__()
+        self.self = BertSelfAttention(
```
Comment (MARD1NO) [inline bert-oneflow/modeling.py:125]: self.self -> self.self_attention

File: bert-oneflow/modeling.py (hunk 1 above)
Comment (L1aoXingyu) [reply bert-oneflow/modeling.py:24]: 这是为了和 lazy 里面 layernorm 名字对齐
https://github.com/Oneflow-Inc/OneFlow-Benchmark/blob/102ab268f3f7bec04fe24c58045a670a9699051d/LanguageModeling/BERT/bert.py#L264

File: bert-oneflow/modeling.py (hunk 3 above)
Comment (L1aoXingyu) [reply bert-oneflow/modeling.py:125]: 这是为了和 lazy 的 self-attention 名字对齐
https://github.com/Oneflow-Inc/OneFlow-Benchmark/blob/102ab268f3f7bec04fe24c58045a670a9699051d/LanguageModeling/BERT/bert.py#L113-L124

**Comment by L1aoXingyu** [comment]
根据 @ShawnXuan xuan哥的建议，使用一个更大的数据集 `/dataset/bert_regression_test/0/part-0`，训练配置如下

```
batch size: 32
iter_num: 1000
lr: 1e-4
optimizer: Adam
scheduler: None
```

前100次Loss
Lazy
```
11.063601
12.700080
11.279208
9.865293
9.909560
9.332666
9.058290
9.221308
8.457822
8.620228
9.074460
8.762793
8.349064
8.632517
8.327777
8.410769
8.604195
8.608268
8.335501
8.263966
8.424753
8.562931
8.527040
8.441665
8.513258
8.269368
8.132557
8.306232
8.040122
8.393031
8.104175
8.506467
8.349229
8.148700
8.050288
8.122770
8.520699
8.101644
8.031592
8.122129
8.340226
8.187310
8.220476
8.133545
8.328325
8.669559
8.383219
8.091852
8.017437
8.360091
8.015594
8.129560
7.941169
8.278321
8.085194
8.165195
8.151962
8.019764
7.964593
7.934909
7.954090
8.091803
8.120596
8.493402
7.915517
7.833041
8.057085
7.959823
7.966826
8.116598
7.982481
8.427887
7.931931
8.027340
7.980044
8.148066
8.232690
8.245623
7.882924
8.058987
7.947914
8.085474
8.165517
8.213579
7.957120
7.944186
8.112403
7.768773
7.850126
8.125142
7.968982
8.154805
7.997802
8.119979
7.927259
7.826028
7.798238
8.161118
7.851645
7.952350
```
Graph
```
11.063609
12.703302
11.300851
9.888453
9.950762
9.380823
9.090010
9.269073
8.528227
8.659925
9.114629
8.799066
8.367892
8.676825
8.349614
8.427313
8.618262
8.631516
8.345539
8.261376
8.434383
8.566625
8.546114
8.458080
8.526626
8.273822
8.134649
8.304640
8.055752
8.397951
8.103891
8.514692
8.362985
8.155717
8.057740
8.125979
8.525864
8.110525
8.038445
8.122572
8.337264
8.189435
8.223833
8.131827
8.329204
8.678854
8.383368
8.095815
8.017931
8.357062
8.014931
8.130983
7.939503
8.270340
8.090549
8.166702
8.153029
8.022255
7.967743
7.940017
7.947827
8.092553
8.115725
8.491781
7.916437
7.832301
8.057820
7.962372
7.970507
8.112662
7.991248
8.422956
7.938228
8.030160
7.976342
8.148544
8.237366
8.244104
7.885255
8.061303
7.948265
8.085780
8.164858
8.218946
7.951743
7.945029
8.119067
7.767087
7.851863
8.126863
7.970669
8.150249
7.998841
8.118787
7.925064
7.822351
7.792827
8.160055
7.851137
7.945285
```

### 1000轮loss曲线
lazy_graph_loss

**Review by ShawnXuan — APPROVED** [review]
